src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -32,9 +33,9 @@ public DeleteCommand(IcNumber icNumber) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Patient> currentPatientList = model.getCurrentPatientList();
+        List<Patient> lastShownList = model.getFilteredPatientList();
```

Comment:
Perhaps it's better to be using the `getCurrentPatientList()` instead, so that you can edit patients outside of the currently visible list? Otherwise the user will have to `list` before editing. This is the behavior for the `delete` command as well. If we want to do it with `getFilteredPatientList()` then we should standardize across all the commands.

Reply:
Yup like what @wujy28 mentioned it would be good to use the `getCurrentPatientList()` as well